src/Build/BackEnd/Components/RequestBuilder/RequestBuilder.cs

```diff
@@ -1118,10 +1118,22 @@ private void SetProjectCurrentDirectory()
         /// </summary>
         private async Task<BuildResult> BuildProject()
         {
+            bool isRestore = false;
+            var propertyEntry = _requestEntry.RequestConfiguration.GlobalProperties[MSBuildConstants.MSBuildIsRestoring];
+            if (propertyEntry != null)
+            {
+                isRestore = Convert.ToBoolean(propertyEntry.EvaluatedValue);
+            }
+
             // We consider this the entrypoint for the project build for purposes of BuildCheck processing 
+            IBuildCheckManager buildCheckManager = null;
 
-            var buildCheckManager = (_componentHost.GetComponent(BuildComponentType.BuildCheck) as IBuildCheckManagerProvider)!.Instance;
-            buildCheckManager.SetDataSource(BuildCheckDataSource.BuildExecution);
+            if (!isRestore)
+            {
+                buildCheckManager = (_componentHost.GetComponent(BuildComponentType.BuildCheck) as IBuildCheckManagerProvider)!.Instance;
+                buildCheckManager.isRestore = false;
+                buildCheckManager.SetDataSource(BuildCheckDataSource.BuildExecution);
+            }
```

Comment:
```suggestion
            IBuildCheckManager buildCheckManager = isRestore ? null : (_componentHost.GetComponent(BuildComponentType.BuildCheck) as IBuildCheckManagerProvider)!.Instance;
            buildCheckManager?.SetDataSource(BuildCheckDataSource.BuildExecution);
```

Similarly all the conditions below can be simplified with null check operator.